### DISCUSSION TITLE
Crashing? Try these things

### DISCUSSION THREAD

**Discussion by knucracker** [KnuckleCracker/CW4-bug-tracker#579] [Ideas]
If the game crashes to desktop, then you will want to look at the log files for the game after a crash. The file is called Player.log and can be found here: 
**`c:\Users\YOUR_USER\AppData\LocalLow\Knuckle Cracker\Creeper World 4\Player.log`** 
and
**`c:\Users\YOUR_USER\AppData\LocalLow\Knuckle Cracker\Creeper World 4\Player-prev.log`** 
Each time your run the game the Player.log file will be overwritten. The Player-prev.log file stores the last log file from the previous run.

Look in the log file. If you see something about sound or graphics, that's your first hint.  You may see a line like this:
`d3d11: failed to create buffer (target 0x5 mode 0 size 144) [0x887A0005]`

That's a problem that starts in the driver (from a software perspective) and then moves up. It can originate with hardware, other software, interaction with other software, driver, etc.

The usual recommended approach is to start conservatively and move outward. So I'd start with:

- **A full reboot**. This resolves the problem like 80% of the time. Do it. You might be surprised.
- **Disable the steam overlay**. Right click the game in steam and go to properties. Disable the steam overlay from there.
- **Other Software**. Ensure other software that might interface with the GPU isn't running. Things like game recording software can be a culprit. Other software like Razor synapse has also been historically associated with a number of game crashes.
- **Reinstall the game.**
- **Run the game in different sizes**. In a window, in a smaller window, full screen, fullscreen exclusive. Do any of those work? That can help narrow down the problem if some work and some don't.
- **Multiple Monitors**. If you are on a multi monitor system try turning the other monitors on or off or running the game on a different monitor. Multi monitor support and some versions of GPU drivers have a higher crash rate.
- **Vulkan Graphics Driver**. Try using a different game graphics driver. The game is compiled with support for Vulkan as well as DirectX11. In steam you can specify a command line argument. Right click the game in steam and go to properties. From there specify the is 'Launch Option': "-force-vulkan" (without the quotes). There may be a few graphical glitches in the game when running under the Vulkan driver. But if it works under Vulkan but crashes under directX11, that can shed some light.
- **Beta Game Build**. New builds of the game (over time) might change unity versions and that could change when and how the crash occurs. So if there is a beta build of the game available on steam, switch to it to see if that affects the problem. Right click the game in steam, go to properties. Then the betas section. From there, you will find a dropdown with "beta" in it. Note that this version might have new features, things in progress, etc.
- **If none of that works**, the problem lies outside the reach of the game. Googling the error in the log will usually show a range of games on a range of machines that have experienced the same problem. The solutions vary, but are outside the scope I can recommend as they involve system changes and other lower level things.

The **Knuckle Cracker Discord** also has a technical_support channel. There may be more info and help from the community available there.
https://discord.gg/knucklecracker

**Comment by piotrj3** [comment]
- **Crash of entire computer** - If your entire computer crash (especially with Blue screen of death) that is not issue of game, issue has to be somewhere in hardware or drivers,
- **Drivers** - in a lot of cases reinstalling/updating drivers help, especially in case of graphics drivers,
- **Other software** - to add on top of Knucracker wrote, especially error prone are bloat controlling RGB/mouses/keyboards/audio (I am not talking about audio drivers, those are necessery!). Ideally to best eliminate 3rd party software issues, system should only have drivers from Chipset, Graphics card, and audio (so in definite majority of cases, only Intel, AMD, Nvidia and Realtek drivers).
- **Overclocking** - in case of issues, it is recommended to run game on stock or decrease overclocking. Occasional Bluescreens might indicate that game crashes from similar instability as system does, however in case of graphics cards, instability might only crash the game itself. In that case verifing that graphics card is stable in MSI kombustor/3d Mark/other GPU intensive games might be helpful.
- **Antivirus** - so far there hasn't been any issues with antiviruses, but they are generally known to cause issues in case of a lot of programs.
